utils/dataset_manifest/core.py

```diff
@@ -0,0 +1,422 @@
+        self._is_created = is_created
+
+    @property
+    def path(self):
+        return self._path
+
+    @property
+    def is_created(self):
+        return self._is_created
+
+    @is_created.setter
+    def is_created(self, value):
+        assert isinstance(value, bool)
+        self._is_created = value
+
+# Needed for faster iteration over the manifest file, will be generated to work inside CVAT
+# and will not be generated when manually creating a manifest
+class _Index:
+    FILE_NAME = 'index'
+
+    def __init__(self, path):
+        assert path and os.path.isdir(path), 'No index directory path'
+        self._path = os.path.join(path, self.FILE_NAME)
+        self._index = {}
+
+    @property
+    def path(self):
+        return self._path
+
+    def dump(self):
+        with open(self._path, 'wb') as index_file:
+            marshal.dump(self._index, index_file, 4)
+
+    def load(self):
+        with open(self._path, 'rb') as index_file:
+            self._index = marshal.load(index_file)
+
+    def create(self, manifest, skip):
+        assert os.path.exists(manifest), 'A manifest file not exists, index cannot be created'
+        with open(manifest, 'r+') as manifest_file:
+            while skip:
+                manifest_file.readline()
+                skip -= 1
+            image_number = 0
+            self._index[image_number] = manifest_file.tell()
+            line = manifest_file.readline()
+            while line:
+                if line.strip():
+                    image_number += 1
+                    self._index[image_number] = manifest_file.tell()
+                line = manifest_file.readline()
+
+    def partial_update(self, manifest, number):
+        assert os.path.exists(manifest), 'A manifest file not exists, index cannot be updated'
+        with open(manifest, 'r+') as manifest_file:
+            manifest_file.seek(self._index[number])
+            line = manifest_file.readline()
+            while line:
+                if line.strip():
+                    self._index[number] = manifest_file.tell()
+                    number += 1
+                line = manifest_file.readline()
+
+    def __getitem__(self, number):
+        assert 0 <= number < len(self), 'A invalid index number'
+        return self._index[number]
+
+    def __len__(self):
+        return len(self._index)
+
+class _ManifestManager(ABC):
+    BASE_INFORMATION = {
+        'version' : 1,
+        'type': 2,
+    }
+    def __init__(self, path, *args, **kwargs):
+        self._manifest = _Manifest(path)
+
+    def _parse_line(self, line):
+        """ Getting a random line from the manifest file """
+        with open(self._manifest.path, 'r') as manifest_file:
+            if isinstance(line, str):
+                assert line in self.BASE_INFORMATION.keys(), \
+                    'An attempt to get non-existent information from the manifest'
+                for _ in range(self.BASE_INFORMATION[line]):
```

Comment:
I thought not, because this is only used to get basic information about the data like `manifest version`, 
`data type` and `video properties`. 
The index is used to get the properties of each image or keyframe and should reduce a chunk creation time.